pkg/sources/github/github.go

```diff
@@ -1478,3 +1500,9 @@ func (s *Source) scanTarget(ctx context.Context, target sources.ChunkingTarget,
 		Verify: s.verify}
 	return handlers.HandleFile(ctx, readCloser, &chunkSkel, reporter)
 }
+
+func (s *Source) ChunkUnit(ctx context.Context, unit sources.SourceUnit, reporter sources.ChunkReporter) error {
+	repoURL, _ := unit.SourceUnitID()
+	ctx = context.WithValue(ctx, "repo", repoURL)
+	return s.scanRepo(ctx, repoURL, reporter)
+}
```

Comment:
I am worried about this, because this implementation only works when called on a source that previously had `Enumerate` called on it, but that precondition is not in any part of the formal or informal `ChunkUnit` contract, so this feels like an accident waiting to happen. I think the rest of this PR is great - is it possible to do this part as a separate, atomic unit of work?

Reply:
Great callout! I think the only change necessary is a call to `ensureRepoInfoCache` before scanning. I've added an integration test that calls `Init()` followed by `ChunkUnit()` to ensure this works.